### PR TITLE
Merge paradox/scala/event-bus.md and java/event-bus.md 

### PR DESCRIPTION
**Pull request by richardimaoka**
Refs #23052

## Rendered HTML:

- https://richard-akka-doc.firebaseapp.com/scala/event-bus.html
- https://richard-akka-doc.firebaseapp.com/java/event-bus.html

## I split this in two commits:

- **1st commit:** Identical contents in scala/event-bus.md and java/event-bus.md -> **this is to take the diff both in scala/event-bus.md and java/event-bus.md**

  - [scala/event-bus.md  (prev) vs. scala/event-bus.md (after)](https://github.com/richard-imaoka/akka/commit/1f5301898f08b0d411c7d9f24261bcf8a0245287#diff-4e2170d1d15e48c56bf6a7a7193737672)
  - [java/event-bus.md (prev) vs. java/event-bus.md (after)](https://github.com/richard-imaoka/akka/commit/1f5301898f08b0d411c7d9f24261bcf8a0245287#diff-2c111bcff1d2c9e09d339b1c4954369a)

- **2nd commit:** Replace java/event-bus.md with a symlink to scala/event-bus.md

### CODE REVIEW

**Comment by richard-imaoka** [comment]
Weird. "Files changed" doesn't show `scala/event-bus.md`? Let me check it...
image

**Comment by raboof** [comment]
It does show `scala/event-bus.md` for me (just shows `java/event-bus.md` only once)

File: akka-docs/src/main/paradox/scala/event-bus.md [hunk 1]
```diff
@@ -185,14 +260,23 @@ and since they are nothing to worry about, they are suppressed from the default
...
+:  @@snip [LoggingDocSpec.scala]($code$/scala/docs/event/LoggingDocSpec.scala) { #all-deadletters }
+
+Java
+:  @@snip [LoggingDocTest.java]($code$/java/jdocs/event/LoggingDocTest.java) { #all-deadletters }
 
 ### Other Uses
 
 The event stream is always there and ready to be used, just publish your own
-events (it accepts `AnyRef`) and subscribe listeners to the corresponding JVM
+events (it accepts `Object`) and subscribe listeners to the corresponding JVM
```
Comment (raboof) [inline akka-docs/src/main/paradox/scala/event-bus.md]: This line needs to be removed

File: akka-docs/src/main/paradox/scala/event-bus.md [hunk 2]
```diff
@@ -126,15 +168,38 @@ carrying @ref:[log messages](logging.md) and [Dead Letters](#dead-letters) and m
 used by the user code for other purposes as well. It uses [Subchannel
 Classification](#subchannel-classification) which enables registering to related sets of channels (as is
 used for `RemotingLifecycleEvent`). The following example demonstrates
+how a simple subscription works. Given a simple actor:
```
Comment (raboof) [inline akka-docs/src/main/paradox/scala/event-bus.md]: 'how a simple subscription works' is duplicated

File: akka-docs/src/main/paradox/scala/event-bus.md [hunk 3]
```diff
@@ -64,11 +81,20 @@ classifier hierarchy.
...
+Scala
+:  @@snip [EventBusDocSpec.scala]($code$/scala/docs/event/EventBusDocSpec.scala) { #subchannel-bus }
+
+Java
+:  @@snip [EventBusDocTest.java]($code$/java/jdocs/event/EventBusDocTest.java) { #subchannel-bus }
 
 A test for this implementation may look like this:
 
-@@snip [EventBusDocSpec.scala]($code$/scala/docs/event/EventBusDocSpec.scala) { #subchannel-bus-test }
+@@snip [EventBusDocTest.java]($code$/java/jdocs/event/EventBusDocTest.java) { #subchannel-bus-test }
```
Comment (raboof) [inline akka-docs/src/main/paradox/scala/event-bus.md]: This one needs to be removed

File: akka-docs/src/main/paradox/scala/event-bus.md [hunk 4]
```diff
@@ -17,11 +22,15 @@ you have to provide it inside the message.
...
-An event bus must define the following three abstract types:
+An event bus must define the following three type parameters:
+An event bus must define the following three @scala[abstract types]@java[type parameters]:
 
+ * `Event` (E) is the type of all events published on that bus
+ * `Subscriber` (S) is the type of subscribers allowed to register on that
  * `Event` is the type of all events published on that bus
  * `Subscriber` is the type of subscribers allowed to register on that
 event bus
+ * `Classifier` (C) defines the classifier to be used in selecting
```
Comment (raboof) [inline akka-docs/src/main/paradox/scala/event-bus.md]: These types now all show up twice in the list

**Review by raboof — CHANGES_REQUESTED** [review]
Some comments, almost there!

**Comment by richard-imaoka** [comment]
Fixed the ones pointed out in the review but still the "Files changed" tab doesn't show `scala/event-bus.md` ...  :expressionless: which is not friendly for review. I still don't know why.

**Comment by raboof** [comment]
Hmm yeah that's weird - viewing individual commits seems to work fine though.

**Pull request merged**